BRD - Apex Arrow suggestions

yumiyafangirl opened the pull request:
Nothing super fancy. It catches some but not all of the possible Apex misuses, including ghosts and uses that were very obviously too low gauge. This action has a lot of nuance in when it should be used and it's impossible for us to accurately predict gauge, so this module will by no means catch ALL bad uses, but it should catch the most egregious ones at least.

![image](https://user-images.githubusercontent.com/65056303/89711330-ac0aea00-d957-11ea-9563-db38161c98e4.png)

suShirow commented:
Did anything here change due to 5.3?

yumiyafangirl commented:
> Did anything here change due to 5.3?

Nope, apex is unchanged.

suShirow commented on src/parser/jobs/brd/modules/ApexArrow.tsx:
```diff
@@ -0,0 +1,108 @@
+				Apex Arrow dealt very low damage <Plural value={this.badApexCasts} one="# time" other="# times"/>.
+			</Trans>,
+			tiers: {
+				1: SEVERITY.MINOR,
```
Isn't just 1 misuse a massive waste of at least medium? How much less damage would it do? Almost as bad as if it were just ghosted. I don't know bard very well though, ignore if this has been thought through already ^^

suShirow commented on src/parser/jobs/brd/modules/ApexArrow.tsx:
```diff
@@ -0,0 +1,108 @@
+	private badApexCasts: number = 0
+
+	protected init() {
+		this.addEventHook('normaliseddamage', {by: 'player', abilityId: ACTIONS.APEX_ARROW.id}, this.onApex)
```
I recommend calling 
`	@dependency private data!: Data`
and then using
`this.data.actions.APEX_ARROW.id` for all uses of ACTIONS.xyz in order to start supporting layered data. Especially since this is a new module, better to keep up!

yumiyafangirl replied:
Thanks, I didn't even know this was necessary in order to make use of the layers.

yumiyafangirl commented:
I added a little more to the debug output in onApex() so we can better track down issues with the potency calc if they ever arise. Here's what the output looks like in a dungeon log where I know apex was always used at 90+ gauge: 

![image](https://user-images.githubusercontent.com/65056303/90343298-2fbe7980-dfdd-11ea-9022-485dc962e174.png)

Max potency is 600, so it's looking reasonably accurate to me. I also confirmed on a handful of known bad apex casts. This module is still too lenient, but I think it's better to err on the safe side and only make a suggestion when the potency was **very** low (lower than a filler GCD). Accordingly, I agree that it makes sense to start the severity at medium for 1 bad use, and I've added this into the PR.
